catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/SecureHadoopCatalogOperations.java

```diff
@@ -196,11 +149,12 @@ public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmpty
       Map<String, String> properties =
           Optional.ofNullable(schemaEntity.properties()).orElse(Collections.emptyMap());
 
-      // Reset the current user based on the name identifier.
-      UserGroupInformation user = getUGIByIdent(properties, ident);
+      UserContext userContext =
+          UserContext.getUserContext(
+              ident, properties, null, hadoopCatalogOperations.getCatalogInfo());
+      boolean r = userContext.doAs(() -> hadoopCatalogOperations.dropSchema(ident, cascade), ident);
+      UserContext.clearUserContext(ident);
```

Comment:
Why not use `userContext.close`?

Reply:
`userContext.close` only close the userConetxt itself and will not remove it from the `userContextMap`